master/docs/developer/utils.rst

```diff
@@ -886,6 +886,44 @@ For example, a particular daily scheduler could be configured on multiple master
         Therefore, in this method it is safe to reassign the "active" status to another instance.
         This method may return a Deferred.
 
+.. py:class:: SingletonService
+
+    This class implements a generic Service that needs to be instantiated as a singleton according to its parameters.
```

Comment:
ok, but I advice to be very careful in defining it in the class doc for programmers. Of course, users do not care about this detail. But, for most programmers, Singleton "is a design pattern that restricts the instantiation of a class to one object", "An implementation of the singleton pattern must: ensure that only one instance of the singleton class ever exists; and provide global access to that instance." [wikipedia](https://en.wikipedia.org/wiki/Singleton_pattern)